cli/cook/querying.py

```diff
@@ -190,19 +227,91 @@ def __get_latest_instance(job):
+    the following shape:
+
+      {'cluster': ..., 'type': ..., 'uuid': ...}
+
+    The cluster field is the name of a cluster, the type field can be either job, instance, or group,
+    and the uuid field is the uuid of the entity in question.
+
+    An entity ref string can either be simply a UUID, in which case all configured clusters and all
+    three types will be queried, or it can be a Cook Scheduler URL that represents the way to retrieve
+    that specific entity.
+
+    Some examples of valid entity ref strings follow:
+
+      5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster2.example.com/jobs/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster2.example.com/jobs?uuid=5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster3.example.com/instances/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster3.example.com/instances?uuid=5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster4.example.com/groups/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster4.example.com/groups?uuid=5ab383b1-5b8b-4483-b2b2-45126923f4df
+
+    Throws if an invalid entity ref string is encountered.
+    """
+    entity_refs = []
+    for ref_string in ref_strings:
+        result = urlparse(ref_string)
+        if result.path and not result.netloc:
+            if is_valid_uuid(result.path):
+                entity_refs.append({'cluster': Clusters.ALL, 'type': Types.ALL, 'uuid': result.path})
+            else:
+                raise Exception(f'{result.path} is not a valid UUID.')
+        elif result.path and result.netloc:
+            path_parts = result.path.split('/')
+            num_path_parts = len(path_parts)
+            if num_path_parts >= 2:
+                cluster_url = (f'{result.scheme}://' if result.scheme else '') + result.netloc
+                cluster_names = [c['name'] for c in clusters if c['url'].lower() == cluster_url.lower()]
+                if len(cluster_names) >= 1:
+                    resource = path_parts[1].lower()
+                    if resource == 'jobs':
+                        entity_type = Types.JOB
+                    elif resource == 'instances':
+                        entity_type = Types.INSTANCE
+                    elif resource == 'groups':
+                        entity_type = Types.GROUP
+                    else:
+                        raise Exception(f'{ref_string} refers to an unsupported resource.')
+
+                    cluster_name = cluster_names[0]
+                    if num_path_parts > 2:
+                        entity_refs.append({'cluster': cluster_name, 'type': entity_type, 'uuid': path_parts[2]})
+                    elif result.query:
+                        query_args = parse_qs(result.query)
+                        if 'uuid' in query_args:
+                            for uuid in query_args['uuid']:
+                                entity_refs.append({'cluster': cluster_name, 'type': entity_type, 'uuid': uuid})
```

Comment:
It's not clear to me if we actually want this branch. (Same concern as with the sample URLs in the docstring.)